openmls/src/credentials/mod.rs

```diff
@@ -133,23 +237,40 @@ impl Credential {
         match &self.credential {
             MlsCredentialType::Basic(basic_credential) => basic_credential.identity.as_slice(),
             // TODO: implement getter for identity for X509 certificates. See issue #134.
-            MlsCredentialType::X509(_) => panic!("X509 certificates are not yet implemented."),
+            MlsCredentialType::X509(certificate_chain) => {
+                // TODO: (wire) implement x509 properly. Identity should be extracted from leaf certificate e.g. serial + subj alt name
+                certificate_chain.identity.as_slice()
+            }
         }
     }
 
     /// Returns the signature scheme used by the credential.
     pub fn signature_scheme(&self) -> SignatureScheme {
         match &self.credential {
             MlsCredentialType::Basic(basic_credential) => basic_credential.signature_scheme,
-            // TODO: implement getter for signature scheme for X509 certificates. See issue #134.
-            MlsCredentialType::X509(_) => panic!("X509 certificates are not yet implemented."),
+            MlsCredentialType::X509(certificate_chain) => {
+                // TODO: implement getter for signature scheme for X509 certificates. See issue #134.
+                // TODO: (wire) highly inefficient, parsing certificate twice to avoid propagating lifetime everywhere
+                let leaf = Certificate::parse_single(certificate_chain.leaf_certificate()).unwrap();
```

Comment:
See: https://docs.rs/x509-parser/latest/x509_parser/certificate/struct.TbsCertificate.html#structfield.signature
Might be able to get the signature scheme and match it against something else to avoid double parsing/lifetime issues

Reply:
I actually might be able to avoid parsing twice if I had the fixed index in the DER document where the signature algorithm is located.  
I went for the simple solution here because anyway it's highly doubtful this PR code gets merged in openmls. WDYT ?